core/src/test/java/org/mobilitydata/gtfsvalidator/util/ServicePeriodTest.java

```diff
@@ -0,0 +1,133 @@
+        .isEqualTo(0b1111111);
+  }
+
+  @Test
+  public void isIncludedInPattern() {
+    // Test all days in 0b0100111 pattern.
+    assertThat(
+        ServicePeriod.isIncludedInPattern(DayOfWeek.MONDAY, (byte)0b0100111))
+        .isTrue();
+    assertThat(
+        ServicePeriod.isIncludedInPattern(DayOfWeek.TUESDAY, (byte)0b0100111))
+        .isTrue();
+    assertThat(
+        ServicePeriod.isIncludedInPattern(DayOfWeek.WEDNESDAY, (byte)0b0100111))
+        .isTrue();
+    assertThat(
+        ServicePeriod.isIncludedInPattern(DayOfWeek.THURSDAY, (byte)0b0100111))
+        .isFalse();
+    assertThat(
+        ServicePeriod.isIncludedInPattern(DayOfWeek.FRIDAY, (byte)0b0100111))
+        .isFalse();
+    assertThat(
+        ServicePeriod.isIncludedInPattern(DayOfWeek.SATURDAY, (byte)0b0100111))
+        .isTrue();
+    assertThat(
+        ServicePeriod.isIncludedInPattern(DayOfWeek.SUNDAY, (byte)0b0100111))
+        .isFalse();
+
+    // Test corner cases: all days included and no days included.
+    assertThat(ServicePeriod.isIncludedInPattern(DayOfWeek.MONDAY, (byte)0))
+        .isFalse();
+    assertThat(
+        ServicePeriod.isIncludedInPattern(DayOfWeek.MONDAY, (byte)0b1111111))
+        .isTrue();
+    assertThat(ServicePeriod.isIncludedInPattern(DayOfWeek.SUNDAY, (byte)0))
+        .isFalse();
+    assertThat(
+        ServicePeriod.isIncludedInPattern(DayOfWeek.SUNDAY, (byte)0b1111111))
+        .isTrue();
+  }
+
+  @Test
+  public void segmentAddedAndRemovedDays() {
+    final LocalDate serviceStart = LocalDate.of(2021, 1, 4);
+    final LocalDate serviceEnd = LocalDate.of(2021, 1, 6);
+    final byte weeklyPattern = 0b1111111;
+    final SortedSet<LocalDate> addedDays = ImmutableSortedSet.of(
+        LocalDate.of(2021, 1, 6), LocalDate.of(2021, 1, 9),
+        LocalDate.of(2021, 1, 10));
+    final SortedSet<LocalDate> removedDays = ImmutableSortedSet.of(
+        LocalDate.of(2021, 1, 5), LocalDate.of(2021, 1, 9));
+    ServicePeriod period = new ServicePeriod(
+        serviceStart, serviceEnd, weeklyPattern, addedDays, removedDays);
+    assertThat(period.getServiceStart()).isEqualTo(serviceStart);
+    assertThat(period.getServiceEnd()).isEqualTo(serviceEnd);
+    assertThat(period.getWeeklyPattern()).isEqualTo(weeklyPattern);
+    assertThat(period.getAddedDays()).isEqualTo(addedDays);
+    assertThat(period.getRemovedDays()).isEqualTo(removedDays);
+    assertThat(period.toDates())
+        .isEqualTo(ImmutableSortedSet.of(LocalDate.of(2021, 1, 4),
+                                         LocalDate.of(2021, 1, 6),
+                                         LocalDate.of(2021, 1, 10)));
+  }
+
+  @Test
+  public void addedDays() {
+    final SortedSet<LocalDate> dates = ImmutableSortedSet.of(
+        LocalDate.of(2021, 1, 3), LocalDate.of(2021, 1, 6),
+        LocalDate.of(2021, 11, 28));
+    assertThat(new ServicePeriod(dates).toDates()).isEqualTo(dates);
+  }
+
+  @Test
+  public void segment() {
```

Comment:
Suggest adding two more tests:
1. where the pattern removes a day of the week such as Thurs 7 Jan, but then that date is in added set
2. where the pattern adds a day of the week such as Mon 4 Jan, but then that date is in removed set

In other words, test interactions between pattern and added/removed dates.

Reply:
This interaction is tested in segmentAddedAndRemovedDays.